arch/riscv/Kconfig

```diff
@@ -27,6 +27,14 @@ config RISCV_GP
 	  global pointer at program start or earlier than any instruction
 	  using GP relative addressing.
 
+config RISCV_CURRENT_VIA_GP
+	bool "Store current thread pointer into the GP register"
```

Comment:
how about `Global Pointer (GP) register`.
Makes it easier to read and understand what GP is...

Reply:
probably also in the kconfig itself, spell it out.

Reply:
I updated the prompt and the help section to spell out the 'global pointer (GP)' but kept the Kconfig name to be consistent with [`CONFIG_RISCV_GP`](https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_RISCV_GP)